subgraphs/livepeer/schema.graphql

```diff
@@ -0,0 +1,536 @@
+  YIELD
+  BRIDGE
+  GENERIC
+  # Will add more
+}
+
+enum TokenType {
+  MULTIPLE
+  UNKNOWN
+  ERC20
+  ERC721
+  ERC1155
+  BEP20
+  BEP721
+  BEP1155
+  # Will add more
+}
+
+type Token @entity {
+  " Smart contract address of the token "
+  id: Bytes!
+
+  " Name of the token, mirrored from the smart contract "
+  name: String!
+
+  " Symbol of the token, mirrored from the smart contract "
+  symbol: String!
+
+  " The number of decimal places this token uses, default to 18 "
+  decimals: Int!
+
+  " Optional field to track the price of a token, mostly for caching purposes "
+  lastPriceUSD: BigDecimal
+
+  " Optional field to track the block number of the last token price "
+  lastPriceBlockNumber: BigInt
+}
+
+enum RewardTokenType {
+  " For reward tokens awarded to LPs/lenders "
+  DEPOSIT
+
+  " For reward tokens awarded to borrowers "
+  BORROW
+}
+
+type RewardToken @entity {
+  " { Reward token type }-{ Smart contract address of the reward token } "
+  id: Bytes!
+
+  " Reference to the actual token "
+  token: Token!
+
+  " The type of the reward token "
+  type: RewardTokenType!
+}
+
+#############################
+##### Protocol Metadata #####
+#############################
+
+type Protocol @entity {
+  " Smart contract address of the protocol's main contract (Factory, Registry, etc) "
+  id: Bytes!
+
+  " Name of the protocol, including version. e.g. Uniswap v3 "
+  name: String!
+
+  " Slug of protocol, including version. e.g. uniswap-v3 "
+  slug: String!
+
+  " Version of the subgraph schema, in SemVer format (e.g. 1.0.0) "
+  schemaVersion: String!
+
+  " Version of the subgraph implementation, in SemVer format (e.g. 1.0.0) "
+  subgraphVersion: String!
+
+  " Version of the methodology used to compute metrics, loosely based on SemVer format (e.g. 1.0.0) "
+  methodologyVersion: String!
+
+  " The blockchain network this subgraph is indexing on "
+  network: Network!
+
+  " The type of protocol (e.g. DEX, Lending, Yield, etc) "
+  type: ProtocolType!
+
+  ##### Quantitative Data #####
+
+  " Current TVL (Total Value Locked) of the entire protocol "
+  totalValueLockedUSD: BigDecimal!
+
+  " Current PCV (Protocol Controlled Value). Only relevant for protocols with PCV. "
+  protocolControlledValueUSD: BigDecimal
+
+  " Revenue claimed by suppliers to the protocol. LPs on DEXs (e.g. 0.25% of the swap fee in Sushiswap). Depositors on Lending Protocols. NFT sellers on OpenSea. "
+  cumulativeSupplySideRevenueUSD: BigDecimal!
+
+  " Gross revenue for the protocol (revenue claimed by protocol). Examples: AMM protocol fee (Sushi’s 0.05%). OpenSea 10% sell fee. "
+  cumulativeProtocolSideRevenueUSD: BigDecimal!
+
+  " All revenue generated by the protocol. e.g. 0.30% of swap fee in Sushiswap, all yield generated by Yearn. "
+  cumulativeTotalRevenueUSD: BigDecimal!
+
+  _cumulativeTransactionCount: Int!
+
+  " Number of cumulative unique users "
+  cumulativeUniqueUsers: Int!
+
+  " Total number of pools "
+  totalPoolCount: Int!
+
```

Comment:
I have changed lastSnapshotDayID, lastSnapshotHourID, and lastUpdateTimestamp to nullable BigInt values because it gave errors with the previous non-nullable Int values and this fixed it.

Reply:
What errors were you getting? I would say we shouldn't change these fields if we don't have to

Reply:
I would say that `cumulativeTransactionCount` is a candidate for being added. We could also add it to the `Pool` entity, and it should look something like this:

```graphql
" Total number of transactions. Transactions include events triggered by outside users (ie, deposit, withdraw, etc.)
cumulativeTransactionCount: Int!
```

We can make this change only a patch as it is strictly additive and not invasive

Reply:
I think the errors were regarding these values being null, even when I was initializing them as zeros. It was in the beginning so I don't remember the exact error. I'll try to revert these values back and recreate the error and see if I can make it work.